lib/_http_outgoing.js

```diff
@@ -283,17 +285,28 @@ OutgoingMessage.prototype._send = function _send(data, encoding, callback) {
 OutgoingMessage.prototype._writeRaw = _writeRaw;
 function _writeRaw(data, encoding, callback) {
   const conn = this.socket;
-  if (conn && conn.destroyed) {
-    // The socket was destroyed. If we're still trying to write to it,
-    // then we haven't gotten the 'close' event yet.
-    return false;
-  }
 
   if (typeof encoding === 'function') {
     callback = encoding;
     encoding = null;
   }
 
+  if (conn && conn.destroyed) {
+    const err = new ERR_STREAM_DESTROYED('write');
```

Comment:
I think this needs to be:
```js
const err = this.finished 
  ? new ERR_STREAM_WRITE_AFTER_END('write') 
  : new ERR_STREAM_DESTROYED('write');
```

Since write after end has priority over stream destroyed.

Reply:
ping @mcollina Do you agree with the above comment?

Reply:
The above are two changes:

- add a check for  `conn.finished`, which I think should be added.
- change the error for `this.finished`, however I'm not sure how to test that, as https://github.com/nodejs/node/blob/e2f7aec52d83ccf356713872bd0d9074c85fd3d9/lib/_http_outgoing.js#L614-L625 provides that behavior.

Reply:
I cannot find a way to trigger that condition. I think the code is ok as it is right now.

Reply:
@mcollina: e.g:

```js
res.end()
res.writeContinue()
```

Reply:
`res.end()` will flip `.finished`, and they will be catched by 872bd0d9074c85fd3d9/lib/_http_outgoing.js#L614-L625. `writeContinue()` does not close the socket.

Reply:
`res.end()` will flio `.finished`. `writeContinue()` will call `_writeRaw()` directly and not go through `write()`

Reply:
`writeContinue()` does not close the underlining socket, so I don't think that applies.

Reply:
I’m probably confused. I thought it was the write after end condition you were having trouble with?

Reply:
I've added another commit to deal with `writeContinue()` and `res.finished`.